### PR TITLE
[1107] Notify MNOs if there are new requests

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/TprR7Aph/1107-build-automatic-notification-for-mnos-when-they-receive-a-request

### Changes proposed in this pull request

- Notify MNO users if they have new requests
- Emails are sent every weekday at 8am

### Screenshots

image

### Guidance to review

- Ensure there is an MobileNetwork with a user
- Create a request for this mno
- Trigger job
- User should see email

### CODE REVIEW

**Comment by fofr** [comment]
If an MNO has a new request at 10am on 2 Dec, will they get an email about it at 8am on 3 Dec?

If they then do not process that request, will they get another email about the same request at 8am on 4 Dec?

**Comment by asmega** [comment]
I've now updated this so we only email if there have been new requests since the last time we emailed them. It's a bit hard coded but it should do the job and I don't expect this to change drastically.

File: app/jobs/notify_mnos_job.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  def since_last_email
+    now = Time.zone.now
+
+    if now.monday?
+      now - 72.hours
+    else
+      now - 24.hours
+    end
+  end
```
Comment (aldavidson) [inline app/jobs/notify_mnos_job.rb:28]: What about using the `EmailAudit` record to record when the last mail of this type was sent to that user, and then querying that log here? It should be more robust in the case of jobs not running at the expected time for whatever reason, or schedule changes etc

**Review by tonyheadford — APPROVED** [review]
Agree with @aldavidson comment that the `since_last_email` could be a bit more robust but don't want to block this.

File: app/jobs/notify_mnos_job.rb (hunk 1 above)
Comment (asmega) [reply app/jobs/notify_mnos_job.rb]: - I've added this in the last commit.
- I felt like i have rushed this a little so could do with a fine comb.
- It won't be as efficient as the previous implementation due to extra queries it'll have to perform.
- There are some drawbacks to doing it this way:
  - different users from same network may get different emails which could be misleading
  - new users will never have had an email sent this is hard coded to 72 hours ago if this is the case

Comment (aldavidson) [reply app/jobs/notify_mnos_job.rb]: This code looks fine to me :+1: 
I wouldn't worry about the extra queries too much - the job runs once a day in the background at 5am,  so it won't impact much.
It's a fair point about different users getting different emails, but I think that's better than the previous situation. It's likely that the number of requests will have changed in between the email being sent and the user clicking on the interface anyway. 
A default timespan of 72hrs seems fine to me.

**Review by aldavidson — APPROVED** [review]

**Pull request merged**